imports/wasi_http/default_http/request.go

```diff
@@ -0,0 +1,36 @@
+package default_http
+
+import (
+	"context"
+	"log"
+
+	"github.com/stealthrocket/wasi-go/imports/wasi_http/types"
+	"github.com/tetratelabs/wazero/api"
+)
+
+//var request = newHostFunc("request", requestFn,
+//	[]api.ValueType{i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32},
+//	"a", "b", "c", "d", "e", "f", "g", "h", "j", "k", "l", "m", "n", "o")
+
+func requestFn(_ context.Context, mod api.Module, a, b, c, d, e, f, g, h, j, k, l, m, n, o uint32) int32 {
+	return 0
+}
+
+//var handle = newHostFunc("handle", handleFn,
+//	[]api.ValueType{i32, i32, i32, i32, i32, i32, i32, i32},
+//	"a", "b", "c", "d", "e", "f", "g", "h")
+
+func handleFn(_ context.Context, mod api.Module, request, b, c, d, e, f, g, h uint32) int32 {
+	req, ok := types.GetRequest(request)
+	if !ok {
+		log.Printf("Failed to get request: %v", request)
+		return 0
+	}
+	r, err := req.MakeRequest()
+	if err != nil {
+		log.Printf(err.Error())
+		return 0
```

Comment:
Is there a way to report more descriptive error codes to the guest?

Reply:
Unfortunately, there's really not. The method is defined here:

https://github.com/WebAssembly/wasi-http/blob/main/wit/types.wit#L86

In theory we could reserve some numbers in this implementation, but it isn't spec'd out in the current interface.

Reply:
I filed this issue on the spec: https://github.com/WebAssembly/wasi-http/issues/44